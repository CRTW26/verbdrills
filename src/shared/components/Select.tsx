interface Props extends React.SelectHTMLAttributes<HTMLSelectElement> {
  options: any
}

const Select: React.FC<Props> = ({ options, ...rest }) => {
  return (
    <select {...rest}>
      {Array.isArray(options) &&
        options.map((option, index) => (
          <option key={index} value={option.value} />
        ))}
    </select>
  )
}

export default Select
