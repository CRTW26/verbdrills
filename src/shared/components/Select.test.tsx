import '@testing-library/jest-dom'
import { render } from '@testing-library/react'
import Select from './Select'

describe('<Select />', () => {
  test('should render', () => {
    const options = [{ value: 'yes' }, { value: 'no' }]
    const { baseElement } = render(<Select options={options} />)

    expect(baseElement).toMatchSnapshot()
  })
})
