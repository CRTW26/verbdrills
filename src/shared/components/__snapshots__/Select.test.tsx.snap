// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> should render 1`] = `
<body>
  <div>
    <select
      class="jsx-2574848865 select"
    >
      <option
        class="jsx-2574848865"
        value="yes"
      >
        yes
      </option>
      <option
        class="jsx-2574848865"
        value="no"
      >
        no
      </option>
    </select>
  </div>
</body>
`;
